src/net/if_.rs

```diff
@@ -21,73 +21,307 @@ pub fn if_nametoindex<P: ?Sized + NixPath>(name: &P) -> Result<c_uint> {
 libc_bitflags!(
     /// Standard interface flags, used by `getifaddrs`
     pub struct InterfaceFlags: libc::c_int {
+        // bitmask: 0x01
         /// Interface is running. (see
         /// [`netdevice(7)`](http://man7.org/linux/man-pages/man7/netdevice.7.html))
         IFF_UP;
+
+        // bitmask: 0x02
         /// Valid broadcast address set. (see
         /// [`netdevice(7)`](http://man7.org/linux/man-pages/man7/netdevice.7.html))
         IFF_BROADCAST;
+
+        // bitmask: 0x04
         /// Internal debugging flag. (see
         /// [`netdevice(7)`](http://man7.org/linux/man-pages/man7/netdevice.7.html))
         IFF_DEBUG;
+
+        // bitmask: 0x08
         /// Interface is a loopback interface. (see
         /// [`netdevice(7)`](http://man7.org/linux/man-pages/man7/netdevice.7.html))
         IFF_LOOPBACK;
+
+        // bitmask: 0x10
         /// Interface is a point-to-point link. (see
         /// [`netdevice(7)`](http://man7.org/linux/man-pages/man7/netdevice.7.html))
         IFF_POINTOPOINT;
+
+        // bitmask: 0x20
         /// Avoid use of trailers. (see
         /// [`netdevice(7)`](http://man7.org/linux/man-pages/man7/netdevice.7.html))
         #[cfg(any(target_os = "android",
+                  target_os = "bitrig",
                   target_os = "fuchsia",
                   target_os = "ios",
                   target_os = "linux",
                   target_os = "macos",
                   target_os = "netbsd",
-                  target_os = "openbsd"))]
+                  target_os = "openbsd",
+                  target_os = "solaris"))]
         IFF_NOTRAILERS;
+        /// Interface manages own routes.
+        #[cfg(any(target_os = "dragonfly"))]
+        IFF_SMART;
+
+        // bitmask: 0x40
         /// Resources allocated. (see
         /// [`netdevice(7)`](http://man7.org/linux/man-pages/man7/netdevice.7.html))
         #[cfg(any(target_os = "android",
+                  target_os = "bitrig",
                   target_os = "dragonfly",
                   target_os = "fuchsia",
                   target_os = "ios",
                   target_os = "linux",
                   target_os = "macos",
                   target_os = "netbsd",
-                  target_os = "openbsd"))]
+                  target_os = "openbsd",
+                  target_os = "solaris"))]
         IFF_RUNNING;
+        /// Resources allocated.
+        #[cfg(any(target_os = "freebsd"))]
+        IFF_DRV_RUNNING;
```

Comment:
`libc` defines only `IFF_DRV_RUNNING` and `IFF_DRV_OACTIVE` for FreeBSD, so that's what I used. Given I don't really use - let alone know - FreeBSD, I don't quite feel in the position to open a PR on `libc`.

Especially since FreeBSD seems to prefer the `DRV` variant since version 8.1:
https://www.freebsd.org/cgi/man.cgi?query=ifnet

Reply:
The distinction between the `DRV` variants and the non-`DRV` variants only matters for kernel code.  You can read the gory details at https://svnweb.freebsd.org/base?view=revision&revision=148886 .  It's perfectly fine to use the non-`DRV` variants in userland code, especially since their values are identical.  For a cross-platform library like nix, the non-`DRV` variants are better, because they're portable.  The current definitions are at https://svnweb.freebsd.org/base/head/sys/net/if.h?view=markup#l170 .  You can go ahead and submit a PR to libc based on those definitions, even if you don't have a FreeBSD VM to test on.  If you like, CC me as a reviewer on the libc PR.